cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -244,6 +244,40 @@ public void getStatus(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, status);
   }
 
+  /**
+   * Returns status of a type specified by the type{flows,workflows,mapreduce,spark,services,schedules}.
+   */
+  @GET
+  @Path("/apps/{app-id}/versions/{version-id}/{type}/{id}/status")
+  public void getVersionStatus(HttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId,
+                               @PathParam("app-id") String appId,
+                               @PathParam("version-id") String versionId,
+                               @PathParam("type") String type,
+                               @PathParam("id") String id) throws Exception {
+    ApplicationId applicationId = new ApplicationId(namespaceId, appId, versionId);
+    if (type.equals("schedules")) {
+      if (!versionId.equals(ApplicationId.DEFAULT_VERSION) || store.getAllAppVersions(applicationId).size() != 1) {
+        throw new NotImplementedException("schedules status is not implemented for applications with" +
```

Comment:
nit: Capitalize `S` in schedules. Also, better to have a space at the end of the line, and then line break